plugins/rest/rest.go

```diff
@@ -312,3 +315,16 @@ func (c Client) Do(ctx context.Context, method, path string) (*http.Response, er
 
 	return resp, err
 }
+
+func withMaskedAuthorizationHeader(headers http.Header) http.Header {
```

Comment:
Can we make this more generic ? For ex.
```go
func maskHeader(headers http.Header, headerName, headerValue string) http.Header {
     // some logic
}
```

We could also just remove the header instead of redacting the value but no strong opinion on that.

Reply:
We sure could, but given how it's not a public function and currently only masking a single header, I think we could make it generic if/when that's needed.